libobs/util/profiler.h

```diff
@@ -14,44 +14,223 @@ typedef struct profiler_time_entry profiler_time_entry_t;
 /* ------------------------------------------------------------------------- */
 /* Profiling */
 
+/*!
+ * \brief Register a new named root Scope.
+ * 
+ * A named root is used to differentiate between profiling sessions. For 
```

Comment:
The actual purpose of this call is to set an expected time between calls (and enable tracking of time between calls); unless you want to track recurring events (such as video rendering) you don't ever need to call this (even when profiling a different thread)